### PR TITLE
Feature/#276/qrcodes

### PR DESCRIPTION
**Pull request by kellhogs**
Ontime settings, Event data:
Removed: Event URL,
Added: Public URL, Backstage URL

handeled database saving on server side.

modified public and backstage view to represent new data

### CODE REVIEW

File: apps/client/src/features/modals/EventSettingsModal.jsx [hunk 1]
```diff
@@ -130,6 +134,22 @@ export default function SettingsModal() {
               onChange={(event) => handleChange('publicInfo', event.target.value)}
             />
           </div>
+          <div className={style.spacedEntry}>
+            <FormLabel htmlFor='backstagekUrl'>
```
Comment (cpvalente) [inline apps/client/src/features/modals/EventSettingsModal.jsx:138]: typo in backstageUrl

File: apps/client/src/features/modals/EventSettingsModal.jsx [hunk 2]
```diff
@@ -99,19 +103,19 @@ export default function SettingsModal() {
           </div>
           <div className={style.hSeparator}>Additional Screen Info</div>
           <div className={style.spacedEntry}>
-            <FormLabel htmlFor='url'>
-              Event URL
+            <FormLabel htmlFor='pubUrl'>
```
Comment (cpvalente) [inline apps/client/src/features/modals/EventSettingsModal.jsx:106]: Nitpick nr 1: I think we should stick to the same naming convention in each entry. So `publicUrl` here and in the name of the input

File: apps/client/src/features/viewers/backstage/Backstage.tsx [hunk 3]
```diff
@@ -146,30 +141,23 @@ export default function Backstage(props) {
...
+      <div className={showPublicMessage ? 'public-container' : 'public-container public-container--hidden'}>
         <div className='label'>Public message</div>
         <div className='message'>{publ.text}</div>
       </div>
 
       <div className='info'>
         <div className='qr'>
-          {general.url != null && general.url !== '' && (
-            <QRCode value={general.url} size={qrSize} level='L' />
+          {general.backstageUrl != null && general.backstageUrl !== '' && (
```
Comment (cpvalente) [inline apps/client/src/features/viewers/backstage/Backstage.tsx:156]: now that we have typescript we can be more confident about the data type, so

`{general.backstage && (`

File: apps/client/src/features/viewers/public/Public.tsx [hunk 4]
```diff
@@ -106,29 +105,23 @@ export default function Public(props) {
...
+      <div className={showPublicMessage ? 'public-container' : 'public-container public-container--hidden'}>
         <div className='label'>Public message</div>
         <div className='message'>{publ.text}</div>
       </div>
 
       <div className='info'>
         <div className='qr'>
-          {general.url != null && general.url !== '' && (
-            <QRCode value={general.url} size={qrSize} level='L' />
+          {general.publicUrl != null && general.publicUrl !== '' && (
```
Comment (cpvalente) [inline apps/client/src/features/viewers/public/Public.tsx:120]: same here as in the other comment
`{general.publicUrl && (`

**Review by cpvalente — CHANGES_REQUESTED** [review]
You did a great job at tracking the data flow, good stuff! I am impressed

I don't have any major comments, just a few cleanups and nitpicks

File: apps/client/src/features/modals/EventSettingsModal.jsx [hunk 5]
```diff
@@ -99,19 +103,19 @@ export default function SettingsModal() {
...
                 <br />
-                Shown as a QR code in some views
+                QR code to be shown on public screens
               </span>
             </FormLabel>
             <Input
               {...inputProps}
-              name='url'
+              name='publicUrl'
               placeholder='www.onsite.no'
```
Comment (cpvalente) [inline apps/client/src/features/modals/EventSettingsModal.jsx:116]: Wow. This is some old code...

Can you please update this to redirect to the ontime website?

Www.getontime.no

Make sure to update all references

File: apps/client/src/features/table/utils.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ export const makeTable = (headerData, tableData, userFields) => {
   const data = [
     ['Ontime · Schedule Template'],
     ['Event Name', headerData?.title || ''],
-    ['Event URL', headerData?.url || ''],
+    ['Public URL', headerData?.publicUrl || ''],
```
Comment (cpvalente) [inline apps/client/src/features/table/utils.js:42]: Are we not missing a backstage url here?

File: apps/client/src/features/viewers/studio/StudioClock.jsx [hunk 7]
```diff
@@ -46,7 +46,7 @@ export default function StudioClock(props) {
   const [searchParams] = useSearchParams();
   const showSeconds = searchParams.get('seconds');
   formatOptions.showSeconds = Boolean(showSeconds);
-  formatOptions.format = `hh:mm${formatOptions.showSeconds ? 'mm' : ''}`;
+  formatOptions.format = `hh:mm${formatOptions.showSeconds ? ':ss' : ''}`;
```
Comment (cpvalente) [inline apps/client/src/features/viewers/studio/StudioClock.jsx:49]: Great catch!

**Review by cpvalente — CHANGES_REQUESTED** [review]
I am recommending some small changes for now , I will build and run the feature later today. Good stuff

File: apps/client/src/features/table/utils.js (hunk 6 above)
Comment (kellhogs) [reply apps/client/src/features/table/utils.js:42]: i have thought about adding it there but decided against it. Maybe the export is shared with third parties that may not have access to the backstage url.

Comment (cpvalente) [reply apps/client/src/features/table/utils.js:42]: We don't really protect the data in the app. The assumption is that anyone that has access to the page should be entitled to the data

Comment (kellhogs) [reply apps/client/src/features/table/utils.js:42]: alright

**Review by cpvalente — APPROVED** [review]

**Pull request merged**